lib/spack/spack/filesystem_view.py

```diff
@@ -371,6 +371,9 @@ def remove_specs(self, *specs, **kwargs):
         with_dependents = kwargs.get("with_dependents", True)
         with_dependencies = kwargs.get("with_dependencies", False)
 
+        # caller can pass this in, as get_all_specs() is expensive
+        all_specs = kwargs.get("all_specs", None)
```

Comment:
I think here you can:
```suggestion
        all_specs = kwargs.get("all_specs", None) or set(self.get_all_specs())
```
and delete the two lines below.

Reply:
Done.